src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java

```diff
@@ -43,21 +43,52 @@
 @ComponentPropertyType
 public @interface SlingServletFilter {
 
-    /** 
-     * Prefix for every property being generated from the annotations elements (as defined in OSGi 7 Compendium, 112.8.2.1) 
+    /**
+     * Prefix for every property being generated from the annotations elements (as defined in OSGi 7 Compendium, 112.8.2.1)
      */
     static final String PREFIX_ = "sling.filter.";
 
+    /**
+     * The scopes of the filter. If the filter has request scope, it is run once for a request. If the filter has component scope, it is run
+     * once for every included component (rendering).
+     * @return the scope of the filter
+     */
+    SlingServletFilterScope[] scope() default SlingServletFilterScope.REQUEST;
+
     /** 
      * Restrict the filter to paths that match the supplied regular expression. Requires Sling Engine 2.4.0.
      * @return the pattern to restrict the filter
      */
     String pattern() default "";
 
-    /** 
-     * The scopes of the filter. If the filter has request scope, it is run once for a request. If the filter has component scope, it is run
-     * once for every included component (rendering). 
-     * @return the scope of the filter
+    /**
+     * Restrict the filter to paths that match the supplied regular expression. Requires Sling Engine 2.4.0.
+     * @return the pattern to restrict the filter
      */
-    SlingServletFilterScope[] scope() default SlingServletFilterScope.REQUEST;
+    String suffix_pattern() default "";
+
+    /**
+     * Restrict the filter to resources that match any of the the supplied resource types. Requires Sling Engine 2.7.0.
+     * @return the resourceTypes to restrict the filter
+     */
+    String[] resourceTypes() default {};
+
+    /**
+     * Restrict the filter to methods that match request's. Requires Sling Engine 2.7.0.
+     * @return the methods to restrict the filter
+     */
+    String[] methods() default {};
+
+    /**
+     * Restrict the filter to extensions that match request's. Requires Sling Engine 2.7.0.
+     * @return the methods to restrict the filter
+     */
+    String[] extensions() default {};
+
+    /**
+     * Restrict the filter to selectors that match any of the the request's. Requires Sling Engine 2.7.0.
```

Comment:
the the

Reply:
What happens if the request has multiple selectors? How are those exactly matched?